### PR TITLE
Adjust doctor command format and options

### PR DESCRIPTION
**Pull request by zeripath**
The gitea doctor command is a little weird especially since the recreate-table subcommand.

This PR changes the doctor command to be an empty subcommand and adds `gitea doctor check` subcommand.

In this subcommand the requested checks are provided as an argument rather than as options.  The logging is also changed so that the doctor.log is not provided by default - instead that will only be created if you set `--log-file` option. `--verbose` is provided to output this logging to stdout.

`gitea convert` is also moved to be a subcommand as `gitea doctor convert`.

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

File: cmd/doctor.go [hunk 1]
```diff
@@ -27,40 +28,11 @@ import (
 var CmdDoctor = cli.Command{
 	Name:        "doctor",
 	Usage:       "Diagnose and optionally fix problems",
-	Description: "A command to diagnose problems with the current Gitea instance according to the given configuration. Some problems can optionally be fixed by modifying the database or data storage.",
-	Action:      runDoctor,
```
Comment (lunny) [inline cmd/doctor.go:30]: Should we still keep the default action to keep compatible? Otherwise we need to add a break label.

Comment (zeripath) [reply cmd/doctor.go:30]: I can't really see the point of keeping the old format - it's not been well-documented, it's kinda an odd format of a command and keeping it around will likely break other things.

File: docs/content/doc/usage/command-line.en-us.md [hunk 2]
```diff
@@ -369,33 +369,22 @@ This command is idempotent.
 
 ### convert
 
-Converts an existing MySQL database from utf8 to utf8mb4.
+(DEPRECATED: moved to doctor convert in 1.19) Converts an existing MySQL database from utf8 to utf8mb4.
```
Comment (lunny) [inline docs/content/doc/usage/command-line.en-us.md]: Use doctor convert instead.

Comment (zeripath) [reply docs/content/doc/usage/command-line.en-us.md]:
```suggestion
(DEPRECATED: from 1.19 use `gitea doctor convert` instead.) Converts an existing MySQL database from utf8 to utf8mb4.
```

**Review by lunny — APPROVED** [review]

**Comment by silverwind** [comment]
Need to update version in references.

**Comment by silverwind** [comment]
I guess @zeripath is currently inactive, but you could raise another PR in place of this one.

**Comment by wxiaoguang** [comment]
There are some problems for this PR:

* The `func init()` seems unnecessary, in most cases the Gitea instance doesn't need to print that message, so it wastes CPU and memory
* The loop `for i, name := range names` is fragile. I can understand the purpose is to make the command like "./gitea doctor check list default", but, it introduces strange syntax like "./gitea doctor check all default list"

So I proposed a new one:  Improve "gitea doctor" sub-command and fix "help" commands #26072

**Pull request closed without merging**